anchore_engine/services/policy_engine/engine/vulns/providers.py

```diff
@@ -126,6 +142,35 @@ def get_sync_utils(self, sync_configs: Dict[str, SyncConfig]) -> SyncUtilProvide
         """
         ...
 
+    def get_feeds_detached(self) -> List[FeedMetadata]:
+        """
+        Returns all feeds excluding grypedb feed in detached state
+        """
+        feeds = get_all_feeds_detached()
+        return list(
+            filter(lambda feed: feed.name in self.__default_sync_config__.keys(), feeds)
+        )
+
+    def get_feed_groups_detached(
+        self, feed: FeedMetadata
+    ) -> Iterable[FeedGroupMetadata]:
+        """
+        Gets a list of all groups consumed by Legacy provider from specified feed in a detached state
+        """
+        if feed.name not in self.__default_sync_config__.keys():
+            raise ValueError(
+                "Incompatible feed passed to Grype provider get_feed_groups_detached function"
```

Comment:
(nit) lose the hardcoded "Grype" in the error message, you can use `self.__config__name__` to reflect the accurate name. Generally when I raise these errors, I would like to see what the offending and allowed values are. So for instance 
```
raise ValueError("%s feed is not supported by %s provider, supported feeds are %s", feed.name, self.__config__name, list(self.__default_sync_config__.keys()))
```